docs/reference/ingest/processors/uri-parts.asciidoc

```diff
@@ -0,0 +1,85 @@
+[role="xpack"]
+[testenv="basic"]
+[[uri-parts-processor]]
+=== URI parts processor
+++++
+<titleabbrev>URI Parts</titleabbrev>
+++++
+
+The URI parts processor parses a text field containing a Uniform Resource
+Identifier (URI) and writes its components to the specified field.
+Components that may be extracted from the URI include domain, path, fragment,
+port, query, scheme, user info, username, and password.
```

Comment:
I tried to streamline this a little bit. If you decide to keep the original text, I'd remove `text field` so users don't confuse this with our `text` field type.

```suggestion
Parses a Uniform Resource Identifier (URI) string and extracts its components as
an object. This URI object includes properties for the URI's domain, path, fragment,
port, query, scheme, user info, username, and password.
```